components/dashboard/src/projects/ConfigureProject.tsx

```diff
@@ -93,52 +125,121 @@ export default function () {
-    <h1>Configure Project</h1>
-    <p className="text-gray-500 text-center text-base">Fully-automate your project's dev setup. <a className="gp-link" href="https://www.gitpod.io/docs/references/gitpod-yml">Learn more</a></p>
-    <div className="mt-4 w-full flex">
-      <div className="flex-1 m-8">
-        {editorError && <AlertBox className="mb-2">{editorError}</AlertBox>}
-        {!isEditorDisabled && <select className="w-full" defaultValue="" onChange={e => setGitpodYml(e.target.value)}>
-          <option value="" disabled={true}>…</option>
-          {Object.entries(TASKS).map(([ name, value ]) => <option value={value}>{name}</option>)}
-        </select>}
-        {!!dockerfile && <div className="flex justify-center border-b border-gray-200 dark:border-gray-800">
+  return <>
+    <Header title="Configuration" subtitle="View and edit project configuration." />
+    <div className="lg:px-28 px-10 mt-8 flex space-x-4">
+      <div className="w-96 h-96 rounded-xl overflow-hidden relative bg-gray-100 dark:bg-gray-700 flex flex-col">
+        <div className="flex bg-gray-50 dark:bg-gray-800 border-b border-gray-200 dark:border-gray-600 px-6 pt-3">
           <TabMenuItem name=".gitpod.yml" selected={selectedEditor === '.gitpod.yml'} onClick={() => setSelectedEditor('.gitpod.yml')} />
-          <TabMenuItem name=".gitpod.Dockerfile" selected={selectedEditor === '.gitpod.Dockerfile'} onClick={() => setSelectedEditor('.gitpod.Dockerfile')} />
-        </div>}
+          {!!dockerfile && <TabMenuItem name=".gitpod.Dockerfile" selected={selectedEditor === '.gitpod.Dockerfile'} onClick={() => setSelectedEditor('.gitpod.Dockerfile')} />}
+        </div>
+        {editorMessage}
         <Suspense fallback={<div />}>
           {selectedEditor === '.gitpod.yml' &&
-            <MonacoEditor classes="mt-4 w-full h-64" disabled={isEditorDisabled} language="yaml" value={gitpodYml} onChange={setGitpodYml} />}
+            <MonacoEditor classes="w-full flex-grow" disabled={isEditorDisabled} language="yaml" value={gitpodYml} onChange={setGitpodYml} />}
           {selectedEditor === '.gitpod.Dockerfile' &&
-            <MonacoEditor classes="mt-4 w-full h-64" disabled={isEditorDisabled} language="dockerfile" value={dockerfile} onChange={setDockerfile} />}
+            <MonacoEditor classes="w-full flex-grow" disabled={isEditorDisabled} language="dockerfile" value={dockerfile} onChange={setDockerfile} />}
         </Suspense>
-        <div className="mt-2 flex justify-center space-x-2">
-          <button disabled={isEditorDisabled} onClick={buildProject}>Save &amp; Test Configuration</button>
-        </div>
+        {isDetecting && <div className="absolute h-full w-full flex items-center justify-center space-x-2">
+          <img className="h-5 w-5 animate-spin" src={isDark ? SpinnerDark : Spinner} />
+          <span className="font-semibold text-gray-400">Detecting project type ...</span>
```

Comment:
**issue**: When adding an empty repository, this is stuck into detecting loop. Could we use an error message and prompt the user to open a new workspace instead?

Reply:
> When adding an empty repository, this is stuck into detecting loop.

Oh, that's unexpected & sounds like a bug. For me, an empty repository gets "detected" as "no setup found", and immediately gets the default config.

Did you notice any errors in the browser console?

Reply:
> Did you notice any errors in the browser console?

Do not recall but will try again once the new preview env is up and running.